src/table.rs

```diff
@@ -1,5 +1,6 @@
-//! Abstractions for Arrow tables. Useful for dataset IO where data will have geometries and
-//! attributes.
+//! Abstractions for Arrow tables.
+//!
+//! Useful for dataset IO where data will have geometries and attributes.
```

Comment:
As a general statement, I'm not sure how much to focus on table-level APIs vs more streaming APIs. Table-level APIs are easy to export to Python and familiar to Python users who've used GeoPandas, but streaming APIs would integrate better into DataFusion. That's why arrow-rs doesn't even have a Table concept, because they want to encourage the use of streaming APIs.

Reply:
🤔 it's a good question. Over in [stac-arrow](https://github.com/stac-utils/stac-rs/blob/ad8dd910c08682344e71c92f0eea62afbfd0a9f8/stac-arrow/src/lib.rs#L202) I found that a `Table` was a good abstraction to give me a method to find the geometry array in a record batch. If you wanted to do away with `Table`, maybe **geoarrow** could just push a `geometry_column` (and friends) down to its own `RecordBatch`?

Reply:
Yeah, geoarrow would in that case probably have an abstraction for a record batch. But for operations like `join` where you want to support multiple record batches at a time, having a `Table` concept is a bit clearer

Reply:
> Yeah, geoarrow would in that case probably have an abstraction for a record batch

Alternatively, in general the current stance of geoarrow is that it should _not_ have an abstraction for most concepts that are fit well via general Arrow objects. E.g. the Python bindings removed its `Table` class because it should be able to reuse general Arrow Table implementations, either in pyarrow or in [arro3](https://github.com/kylebarron/arro3), and just look for geoarrow metadata.

So similarly, an algorithm like `join` could take in
```
fn join(left: Box<dyn RecordBatchReader>, right: Box<dyn RecordBatchReader>) -> Box<dyn RecordBatchReader>
```
Where the RecordBatchReader is _just_ an arrow-rs type, not anything specific to the geoarrow crate.